### DISCUSSION TITLE
Snapshot Creation

### DISCUSSION THREAD

**Discussion by sajidhashmi** [iTwin/itwinjs-core#126] [Q&A] [answered]
Hi,

I am trying to create a snapshot from iModel file. Found `createFrom()` and `createEmpty()` as two ways to do so. Is there any option to provide a file which is locally saved from which snapshot can be created? 
Also, when I try by sending BriefcaseDb as seedFile, I get the error that `BE_SQLITE_MISUSE: Cannot create a snapshot from an encrypted iModel`. While doing this, I am first downloading Briefcase from iModel using `BriefcaseManager.download()` and then using `BriefcaseDb.open()`. Am I missing any step in  between?

Any help would be appreciated.

**Comment by scsewall** [comment, marked as answer]
The `createFrom()` API takes an `IModelDb` parameter (an open iModel) instead of a file name to ensure that you have permission to open the file before allowing its contents to be copied. You would just call one of the open methods on the seed file before passing the returned `IModelDb` into `createFrom()`.

Your error message suggests that the iModel that you are trying to copy has a password or encryption key protecting its contents. The first step would be to confirm whether or not that is true.
